classes/observers/listeners/MailDiscussionUpdated.inc.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * @file classes/observers/listeners/MailDiscussionUpdate.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class MailDiscussionUpdate
+ * @ingroup observers_listeners
+ *
+ * @brief Notify user about discussion being updated
+ */
+
+namespace PKP\observers\listeners;
+
+use Illuminate\Support\Facades\Mail;
+use PKP\core\PKPContainer;
+use PKP\db\DAORegistry;
+use APP\facades\Repo;
+use PKP\notification\NotificationSubscriptionSettingsDAO;
+use PKP\notification\PKPNotification;
+use PKP\observers\events\DiscussionMessageSent;
+use PKP\mail\mailables\MailDiscussionUpdated as Mailable;
+
+class MailDiscussionUpdated
+{
+    /**
+     * @param \PKP\observers\events\DiscussionMessageSent $event
+     */
+    public function handle(DiscussionMessageSent $event)
+    {
+        $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
+
+        $users = $event->formEmailData->getRecipients($event->context->getId());
+        foreach ($users as $user) {
+            // Check if user is unsubscribed
+            $notificationSubscriptionSettings = $notificationSubscriptionSettingsDao->getNotificationSubscriptionSettings(
+                NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY,
+                $user->getId(),
+                (int) $event->context->getId()
+            );
+            if (in_array(PKPNotification::NOTIFICATION_TYPE_NEW_QUERY, $notificationSubscriptionSettings))
+                return;
+
+            $submission = Repo::submission()->get($event->query->getAssocId());
+            $mailer = PKPContainer::getInstance()['mailer'];
+            $sender = $event->formEmailData->getSender();
+
+            $mailable = new Mailable($event->context, $submission, $sender, [$user]);
+            $emailTemplate = $mailable->getTemplate($event->context->getId());
+
+            $templateData = [
+                'siteTitle' => $mailable->viewData['contextName'],
+            ];
+
+            $templateData = array_merge($templateData, $event->formEmailData->getVariables($user->getId()), $mailable->viewData);
+
+            $body = $mailer->compileParams($emailTemplate->getLocalizedData('body'), $templateData);
+            $subject = $mailer->compileParams($emailTemplate->getLocalizedData('subject'), $templateData);
+            $mailable
+                ->html($body)
+                ->subject($subject)
+                ->from($sender->getEmail(), $sender->getFullName())
+                ->replyTo($event->context->getContactEmail(), $event->context->getContactName())
```

Comment:
Took a closer look. All notification emails have their `reply to` set to either context or site contact email: https://github.com/pkp/pkp-lib/blob/0f727b565929b04ddcbc0620cfa05f7471635364/classes/notification/PKPNotificationOperationManager.inc.php#L429.  @asmecher, do you think we should preserve this behaviour here, for discussions?

Reply:
Discussion emails are not intended to be replied to directly, until OJS/OMP/OPS is able to receive the emails itself and process them (like replying to a Github email). Setting the reply-to address to the sender would encourage receivers to just reply to the email rather than logging into the system to respond via the discussion feature.

Reply:
Understood, thanks!